### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string.

### ISSUE THREAD

**Issue by tymmesyde** [ArekSredzki/electron-release-server#189]
I'm having an issue with the server update query.
My server is fully set up and works nicely with this options on the electron `autoUpdater.setFeedURL` : 
```
let options = {
   url: `${url}/update/${os.platform()}/${app.getVersion()}/stable`
}
autoUpdater.setFeedURL(options);
```

But when i replace `os.platform()` by a string like this :
```
let options = {
   url: `${url}/update/win64/${app.getVersion()}/stable`
}
autoUpdater.setFeedURL(options);
```
I get this :
`
Sending 500 ("Server Error") response:
TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined at Url.parse (url.js:150:11)
`

Otherwise it seems to understand my query before throwing me this error:
```
| Windows Update Search Query { platform: [ 'windows_64' ],
|   version: '1.1.1',
|   channel: 'stable' }
| Applicable Channels [ 'stable' ]
| Time Filter { '>=': 2018-08-06T17:02:37.000Z }
| Latest Windows Version { assets:
|    [ { name: 'SlipBot-1.1.1-full.nupkg',
|        platform: 'windows_64',
|        filetype: '.nupkg',
|        hash: 'A7D1A45796284232D8E54A0A9150F74731D978E5',
|        size: 78692234,
|        download_count: 0,
|        fd:
|         '/var/www/slipbot.idden.me/.tmp/uploads/assets/f5353fdd-d76d-4a47-94c7-6b0fa887ffca.nupkg',
|        createdAt: '2018-08-06T18:09:08.000Z',
|        updatedAt: '2018-08-06T18:09:08.000Z',
|        version: '1.1.1' } ],
|   channel: 'stable',
|   name: '1.1.1',
|   notes: '',
|   createdAt: '2018-08-06T17:02:37.000Z',
|   updatedAt: '2018-08-06T17:02:37.000Z' }
```

I really don't understand whats happening here, can you help me ?

**Comment by Grace1133** [comment]
#171

**Comment by Grace1133** [comment]
#197

**Comment by Grace1133** [comment]
#195

**Comment by Grace1133** [comment]
#171

**Comment by Grace1133** [comment]
#107

**Comment by Grace1133** [comment]
#195

**Comment by andypaxo** [comment]
It's over a year later and the issue has been closed, but this is the first search result for the error, so I'll leave instructions on how to deal with the problem here, as I just ran into the same issue.

This error occurs when `appUrl` is unset in the config. It should be set to the base URL of your release server.
If you're using your own local.config, then this can be set in the file.
If you're using the provided docker.config, then set the `APP_URL` environment variable.
